pkg/providers/ssm/provider.go

```diff
@@ -60,3 +74,54 @@ func (p *DefaultProvider) Get(ctx context.Context, parameter string) (string, er
 	log.FromContext(ctx).WithValues("parameter", parameter, "value", result.Parameter.Value).Info("discovered ssm parameter")
 	return lo.FromPtr(result.Parameter.Value), nil
 }
+
+// Invalidate asyncronously invalidates entries in the underlying SSM cache. Invalidation is scheduled up to 30 minutes
+// in the future to avoid thundering herds caused by globally correlated invalidations. Any subsequent invalidations of
+// the same parameter made before the initial invalidation has been processed are no-ops.
+func (p *DefaultProvider) Invalidate(ctx context.Context, parameters ...string) {
+	p.Lock()
+	defer p.Unlock()
+
+	nonPending := map[string]string{}
```

Comment:
Maybe I'm not tracking but why call this variable nonPending?

Reply:
The naming could change, but this should be any invalidation requests which are not currently pending (e.g. already enqueued in a separate goroutine).